Framework/DataHandling/test/SaveNexusGeometryTest.h

```diff
@@ -94,13 +68,68 @@ class SaveNexusGeometryTest : public CxxTest::TestSuite {
     TS_ASSERT(alg.isInitialized());
 
     TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", "testWS"));
-
     TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("FileName", destinationFile));
     TS_ASSERT_THROWS_NOTHING(
         alg.setPropertyValue("H5Path", "algorithm_test_data"));
     TS_ASSERT_THROWS_NOTHING(alg.execute());
     TS_ASSERT(alg.isExecuted());
   }
+
+  void
+  test_execution_succesful_when_no_h5_root_provided_and_default_root_is_used() {
+
+    ScopedFileHandle fileResource("algorithm_invalid_workspace_file.hdf5");
+    auto destinationFile = fileResource.fullPath();
+    // Create test input if necessary
+    Mantid::API::MatrixWorkspace_sptr inputWS = WorkspaceCreationHelper::
+        create2DDetectorScanWorkspaceWithFullInstrument(1, 5, 1);
+
+    TS_ASSERT_THROWS_NOTHING(
+        Mantid::API::AnalysisDataService::Instance().addOrReplace("testWS",
+                                                                  inputWS));
+    SaveNexusGeometry alg;
+    // Don't put output in ADS by default
+    alg.setChild(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized());
+
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", "testWS"));
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("FileName", destinationFile));
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+  }
+
+  void test_invalid_workspace_throws() {
+
+    ScopedFileHandle fileResource(
+        "algorithm_no_h5_path_provided_test_file.hdf5");
+    auto destinationFile = fileResource.fullPath();
+    // Create test input if necessary
+
+    WorkspaceCreationHelper::EPPTableRow row(
+        1, 1, 1, WorkspaceCreationHelper::EPPTableRow::FitStatus::SUCCESS);
+    std::vector<WorkspaceCreationHelper::EPPTableRow> rows{row};
+    Mantid::API::ITableWorkspace_sptr inputWS =
+        WorkspaceCreationHelper::createEPPTableWorkspace(rows);
+
+    // TS_ASSERT_THROWS_NOTHING(
+    //  Mantid::API::AnalysisDataService::Instance().add("testWS", inputWS));
+    TS_ASSERT_THROWS_NOTHING(
+        Mantid::API::AnalysisDataService::Instance().addOrReplace("testWS",
+                                                                  inputWS));
+    SaveNexusGeometry alg;
+    // Don't put output in ADS by default
+
+    alg.setChild(true);
```

Comment:
Interesting with checking the throw below. I thought that you would have to `alg.setRethrows(true)` to get it the throw the original exception.